src/qml/WebCatalogPage.qml

```diff
@@ -167,19 +183,19 @@ Page {
                     width: parent.width
                     Label {
                         id: nameLabel
-                        width: parent.width - authorLabel.width - Theme.paddingMedium
+                        width: parent.width - secondaryLabel.width - Theme.paddingMedium
                         text: model.display_name
                         color: background.down ? Theme.highlightColor : Theme.primaryColor
                         font.bold: isInstalled
                         truncationMode: TruncationMode.Fade
                     }
 
                     Label {
-                        id: authorLabel
+                        id: secondaryLabel
                         anchors.right: parent.right
                         anchors.verticalCenter: parent.verticalCenter
                         font.pixelSize: Theme.fontSizeExtraSmall
-                        text: model.author
+                        text: container.sortByDate ? model.category : model.author
```

Comment:
The question here is how do you see this in the future:
Right now, you have the equivalent of:
 - no sorting - you see sections == categories, secondary label == author
 - by category - you don't see sections, secondary label = category

How does this 'scale'? What if you introduce
 - by author - you don't see sectons, secondary label = category (?)

Reply:
My thinking was doing away with the boolen completely, introducing an enum or a State in the QML, and using that to steer sorting.

Sorting itself should also be done in the model so it doesn't trigger a new request to the server if changed - as @mentaljam suggested earlier.

But that is something to tackle in the future when an improved version takes shape.